### PR TITLE
SONARIAC-1325 S6907: Environment variables for a container should not be duplicated

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    containers.clear();
+  }
+
+  private static void reportIssue(CheckContext ctx, List<YamlTree> trees) {
+    var secondaryLocations = trees.stream()
+      .skip(1)
+      .map(t -> new SecondaryLocation(t, MESSAGE_SECONDARY_LOCATION))
+      .toList();
+    ctx.reportIssue(trees.get(0), MESSAGE, secondaryLocations);
+  }
```
[the remark this answers is not in the document]
Comment (mstachniuk) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java:94]: I wasn't sure what would be better for the primary location if there were more duplications. I can change the primary location to the last one.

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/KubernetesChecksVisitor.java [hunk 2]
```diff
@@ -96,6 +96,7 @@ protected void reportIssue(@Nullable TextRange textRange, String message, List<S
         }
         List<SecondaryLocation> shiftedSecondaryLocations = secondaryLocations.stream()
           .map(secondaryLocation -> LocationShifter.computeShiftedSecondaryLocation(helmCtx, secondaryLocation))
+          .distinct()
```
[the remark this answers is not in the document]
Comment (mstachniuk) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/KubernetesChecksVisitor.java:99]: Yes. After shifting locations for the template `{{- .Values.envVariables3 | toYaml | nindent 8 }}` the secondary locations may point to the same line in `values.yaml`, which looks not nice in SQ UI.

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonas-wielage-sonarsource) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java:94]: Maybe it makes also sense to change the message on secondary locations: Duplicate environment variable without any effect? Or something like this

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/KubernetesChecksVisitor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonas-wielage-sonarsource) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/KubernetesChecksVisitor.java:99]: Nice catch

**Review by jonas-wielage-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**